File: iosApp/iosApp/KtorDataModel.swift
//
//  KtorDataModel.swift
//  iosApp
//
//  Created by Thanasis Papapostolou on 05/07/2023.
//  Copyright © 2023 orgName. All rights reserved.
//

import SwiftUI;
import shared;

class KtorDataModel: ObservableObject {
    @Published var works: [CyberWork] = [];
    @Published var library = "Ktor";

    init() {
        KMMCommonKtorJsonGetter().getJson { data, error in
            if let cyberlist = data {
                DispatchQueue.main.async {
                    self.works = cyberlist.cyberpunk_works;
                }
            }
            else {
                let err = error?.localizedDescription ?? "error";
                self.works = [CyberWork(name: err, creator: err, year: err)];
            }
        }
    }
}

File: androidApp/src/main/java/com/example/ktorkmm/android/KtorDataModel.kt
package com.example.ktorkmm.android

import com.example.ktorkmm.CyberList
import com.example.ktorkmm.CyberWork
import com.example.ktorkmm.KMMCommonKtorJsonGetter

class KtorDataModel {
    val library = "Ktor";

    suspend fun getJson(): CyberList {
        return try {
            KMMCommonKtorJsonGetter().getJson();
        }
        catch(e: Exception) {
            val err = e.localizedMessage ?: "error";
            CyberList(listOf(CyberWork(err, err, err)));
        }
    }
}
